go/chisel/share/radius_proxy/tunnel_radius.go

```diff
@@ -0,0 +1,166 @@
+//go:build !test_radius
+// +build !test_radius
+
+package radius_proxy
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/inverse-inc/go-utils/sharedutils"
+	"github.com/inverse-inc/packetfence/go/chisel/share/cio"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+)
+
+const radiusAuthK8Filter = "app=radiusd-auth"
+
+func isPodReady(pod *v1.Pod) bool {
+	if pod.DeletionTimestamp != nil {
+		return false
+	}
+
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == v1.PodReady {
+			return cond.Status == v1.ConditionTrue
+		}
+	}
+
+	return false
+}
+
+func clientSetFromEnv() (*kubernetes.Clientset, error) {
+	host := os.Getenv("K8S_MASTER_URI")
+	if host == "" {
+		return nil, errors.New("K8S_MASTER_URI is not defined")
+	}
+
+	token := os.Getenv("K8S_MASTER_TOKEN")
+	if token == "" {
+		return nil, errors.New("K8_MASTER_TOKEN is not defined")
+	}
+
+	return kubernetes.NewForConfig(
+		&rest.Config{
+			Host:            host,
+			BearerToken:     token,
+			TLSClientConfig: TLSClientConfigFromEnv(),
+		},
+	)
+}
+
+func NewRadiusProxyFromKubernetes(l *cio.Logger, radiusSecret string) (*Proxy, chan struct{}, error) {
+	clientset, err := clientSetFromEnv()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	data, err := os.ReadFile(os.Getenv("K8S_NAMESPACE_PATH"))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	namespace := string(data)
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: radiusAuthK8Filter})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	servers := []string{}
+	for _, p := range pods.Items {
+		addr := p.Status.PodIP + ":1812"
```

Comment:
The pod metadata can expose a containerPort to replace the hardcoded value. Or we should at least allow the environment to override the 1812 value

Reply:
Which name should we use to match in the which Port array?

Reply:
Write now it is called the theport

Reply:
I'm not sure I follow your thoughts